src/main/kotlin/org/rust/ide/annotator/RsHighlightingMutableAnnotator.kt

```diff
@@ -0,0 +1,83 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.lang.annotation.AnnotationHolder
+import com.intellij.lang.annotation.Annotator
+import com.intellij.lang.annotation.HighlightSeverity
+import com.intellij.psi.PsiElement
+import org.rust.ide.colors.RsColor
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsElement
+import org.rust.lang.core.psi.ext.RsNameIdentifierOwner
+import org.rust.lang.core.psi.ext.ancestorStrict
+import org.rust.lang.core.psi.ext.mutability
+import org.rust.lang.core.types.ty.TyReference
+import org.rust.lang.core.types.type
+import org.rust.openapiext.isUnitTestMode
+
+class RsHighlightingMutableAnnotator : Annotator {
+    private val HIGH_PRIORITY_HIGHLIGHTING = HighlightSeverity(
+        "HIGH_PRIORITY_HIGHLIGHTING",
+        HighlightSeverity.INFORMATION.myVal + 3
+    )
+
+    override fun annotate(element: PsiElement, holder: AnnotationHolder) {
+        val ref = when (element) {
+            is RsPath -> element.reference.resolve() ?: return
+            is RsSelfParameter -> element
+            is RsPatBinding -> element
+            else -> return
+        }
+        distinctAnnotation(element, ref, holder)
+    }
+
+    private fun annotationFor(ref: RsElement): Pair<RsNameIdentifierOwner, RsColor>? = when (ref) {
+        is RsSelfParameter -> ref to RsColor.MUT_PARAMETER
+        is RsPatBinding -> if (ref.ancestorStrict<RsValueParameter>() != null) {
+            ref to RsColor.MUT_PARAMETER
+        } else {
+            ref to RsColor.MUT_BINDING
+        }
+        else -> null
+    }
+
+    private fun distinctAnnotation(element: PsiElement, ref: RsElement, holder: AnnotationHolder) {
+        val (ref, color) = annotationFor(ref) ?: return
+        if (ref.isMut) {
+            val element = partToHighlight(element)
+            addHighlightingAnnotation(holder, element, color)
+        }
+    }
+
+    private fun partToHighlight(element: PsiElement): PsiElement {
+        val element = when (element) {
+            is RsSelfParameter -> element.self
+            is RsPatBinding -> element.identifier
+            else -> element
+        }
+        return element
+    }
+
+    private fun addHighlightingAnnotation(holder: AnnotationHolder, target: PsiElement, key: RsColor) {
+        // These following lines allow to test the mutable highlight for it own using WEAK_WARNING and the description.
+        val annotationSeverity = if (isUnitTestMode) HighlightSeverity.WEAK_WARNING else HIGH_PRIORITY_HIGHLIGHTING
```

Comment:
Let's use `HighlightSeverity.INFORMATION` for tests because it's more suitable for this annotation and allows fixing tests in `RsHighlightingAnnotatorTest` without unnecessary changes.
Don't forget to correct comment too :wink: